libff/algebra/curves/tests/test_groups.cpp

```diff
@@ -62,53 +88,52 @@ void test_group()
     bigint<1> randsum = bigint<1>("121160274");
 
     GroupT zero = GroupT::zero();
-    assert(zero == zero);
+    EXPECT_EQ(zero, zero);
     GroupT one = GroupT::one();
-    assert(one == one);
+    EXPECT_EQ(one, one);
     GroupT two = bigint<1>(2l) * GroupT::one();
-    assert(two == two);
+    EXPECT_EQ(two, two);
```

Comment:
Aren't the line above tautological? See: https://github.com/scipr-lab/libff/pull/38

Reply:
Oh! I think I understand what @ValarDragon said on my PR. The `one == one` is a way to _test_ the `operator==`, is that what you meant @ValarDragon?
I was expecting a separate test for that... i.e. having a unit test per function (with positive and negative test cases) instead of a test function (like this `test_group`) testing a bunch of functions at the same time.

Reply:
Yeah that is what I meant! Sorry I didn't reply to that

Reply:
No problem, thanks for confirming! I get your point, the asserts I removed in #38 are definitely not "dummy/tautological" asserts. They are test cases, so we need to keep them.

That being said, I find the current test structure quite confusing. There is a `test_mixed_add()` function to test the mixed addition, a `test_mul_by_q()` function to test mult by field characteristic, a `test_output()` function for the `operator>>` and `operator<<`, and the rest of the functions of the "group interface" are tested in this `test_group()` function. I find this quite inconsistent - i.e. some functions have their own tests (1 function - 1 associated test), some don't. Additionally, I am not sure that all functions from the group are actually tested here (I can't see any test for `is_well_formed` for instance).

I personally tend to prefer over verbosity in tests as a way to make it trivial to see what is tested and what is not, and avoid mistakes.

If sticking to a single test `test_group` is desired here, I think doing something like:
```
template<typename GroupT>
void test_group{
    test_add<GroupT>(); // test add() function
    test_is_well_formed<GroupT>(); // test is_well_formed()
    test_dbl<GroupT>(); // test dbl()
    test_equal<GroupT>(); // test operator==() (can also test operator!=() or do it in another fucntion)
    ...
    test_is_zero<GroupT>();
    test_mul_by_q<GroupT>();
}
```

Would be a good improvement, as we would test each function independently, and these tests would be self contained. This would avoid mixing all test cases in one big test function.

Any thoughts?

Reply:
That makes sense, though I never touched this code other than using gtest. Since this seems to require a relatively large refactor, maybe save it for a future PR.

Reply:
I have opened a ticket for that, see: https://github.com/scipr-lab/libff/issues/46

Reply:
This seems like a great change for readability to me as well